src/web/services/feedback-responses.service.spec.ts

```diff
@@ -1,16 +1,429 @@
+      .toBeFalsy();
+  });
+
+  it('should correctly display responses or not for an "evaluee" section type', () => {
+    const response: ResponseOutput = {
+      giverSection: 'giver section',
+      recipientSection: 'recipient section',
+    } as ResponseOutput;
+    let section: string = 'giver section';
+    const sectionType: InstructorSessionResultSectionType = InstructorSessionResultSectionType.EVALUEE;
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeFalsy();
+
+    section = 'recipient section';
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeTruthy();
+
+    section = 'wrong section';
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeFalsy();
+  });
+
+  it('should correctly display responses or not for a "both" section type', () => {
+    const response: ResponseOutput = {
+      giverSection: 'section',
+      recipientSection: 'section',
+    } as ResponseOutput;
+    let section: string = 'section';
+    const sectionType: InstructorSessionResultSectionType = InstructorSessionResultSectionType.BOTH;
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeTruthy();
+
+    section = 'wrong section';
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeFalsy();
+  });
+
+  it('should display responses when no section is specified', () => {
+    const response: ResponseOutput = {
+      giverSection: 'giver section',
+      recipientSection: 'recipient section',
+    } as ResponseOutput;
+    let section: string = '';
+    const sectionType: InstructorSessionResultSectionType = InstructorSessionResultSectionType.EITHER;
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeTruthy();
+
+    section = null as unknown as string;
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeTruthy();
+
+    section = undefined as unknown as string;
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeTruthy();
+  });
+
+  it('should display responses for an unknown section type', () => {
+    const response: ResponseOutput = {
+      giverSection: 'giver section',
+      recipientSection: 'recipient section',
+    } as ResponseOutput;
+    const section: string = 'giver section';
+    const sectionType: InstructorSessionResultSectionType = 'UNKNOWN' as InstructorSessionResultSectionType;
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeTruthy();
+  });
+
+  it('should call get when retrieving a feedback response', () => {
+    const dummyIntent: Intent = Intent.STUDENT_SUBMISSION;
+    const paramMap: Record<string, string> = {
+      questionid: '[auto-generated Datastore ID]',
```

Comment:
Don't make any mention to "Datastore" or anything DB-related here. Even 'dummy question id' is sufficient.